src/Core/Form/IdentifiableObject/DataHandler/ProfileFormDataHandler.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Profile\Command\AddProfileCommand;
+use PrestaShop\PrestaShop\Core\Domain\Profile\Command\EditProfileCommand;
+use PrestaShop\PrestaShop\Core\Domain\Profile\ValueObject\ProfileId;
+
+/**
+ * Saves or updates Profile using form data
+ */
+final class ProfileFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $bus;
+
+    /**
+     * @param CommandBusInterface $bus
+     */
+    public function __construct(CommandBusInterface $bus)
+    {
+        $this->bus = $bus;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $data)
+    {
+        /** @var ProfileId $profileId */
+        $profileId = $this->bus->handle(new AddProfileCommand($data['name']));
+
+        return $profileId->getValue();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($profileId, array $data)
+    {
+        /* @var ProfileId $profileId */
```

Comment:
I don't think this PHPDoc is useful here

Reply:
It helps for autocomplete because method signature doesn't have `Profileid $profileId`